drivers/serial/serial.c

```diff
@@ -854,6 +854,16 @@ static ssize_t uart_read(FAR struct file *filep, FAR char *buffer, size_t buflen
           break;
         }
 
+      else if (filep->f_inode == 0)
```

Comment:
check in uart_write?

Reply:
I think it is not necessary.
The write operation is not susceptible to deadlock. it is just serial, so low level driver is responsible to send all the data eventually.

Reply:
But, it depends on what the follow control will be used. If user enable the hardware RTS/CTS, the caller may block in write for a very long time.

Reply:
The close function has already a logic to wait for clearing transmit buffer. So the close function will block.

/* Prevent blocking if the device is opened with O_NONBLOCK /
if ((filep->f_oflags & O_NONBLOCK) == 0)
{
/ Now we wait for the transmit buffer(s) to clear */
uart_tcdrain(dev, 4 * TICK_PER_SEC);
}

Reply:
Yes, but the close only wait for 4 seconds.

Reply:
No. Timeout will not work. 
It will block on 
**uart_takesem(&dev->xmit.sem, true);**
until the ongoing write function exits. Any other calls to write function after close will fail.

Reply:
Maybe, we should forward the time to uart_takesem too.